src/yaml-support/tkn-code-actions.ts

```diff
@@ -0,0 +1,174 @@
+    return vscode.window.withProgress({location: vscode.ProgressLocation.Notification, cancellable: false, title: `Loading '${codeAction.taskRefName}' Task...` }, async (): Promise<vscode.CodeAction> => {
+      const uri = tektonFSUri(codeAction.taskKind === TektonYamlType.ClusterTask ? ContextType.CLUSTERTASK : ContextType.TASK , codeAction.taskRefName, 'yaml');
+      try {
+        const taskDoc = await tektonVfsProvider.loadTektonDocument(uri, false);
+        codeAction.edit = new vscode.WorkspaceEdit();
+        codeAction.edit.replace(codeAction.documentUri,
+          new vscode.Range(codeAction.taskRefStartPosition, codeAction.taskRefEndPosition), 
+          this.extractTaskDef(taskDoc, codeAction.taskRefStartPosition.character, codeAction.taskRefEndPosition.character));
+      } catch (err){
+        vscode.window.showErrorMessage('Cannot get Tekton Task definition: ' + err.toString());
+        telemetryLogError('resolveCodeAction', `Cannot get '${codeAction.taskRefName}' Task definition`);
+      }
+      return codeAction;
+    });
+    
+  }
+
+  getProviderMetadata(): vscode.CodeActionProviderMetadata {
+    return {
+      providedCodeActionKinds: [vscode.CodeActionKind.RefactorInline.append('TektonTask')],
+    }
+  }
+
+  findTask(doc: TknDocument, pos: vscode.Position): PipelineTask | undefined {
+    const selectedElement = doc.findElementAt(pos);
+    if (!selectedElement) {
+      return undefined;
+    }
+    if (selectedElement.type === TknElementType.PIPELINE_TASK) {
+      return selectedElement as PipelineTask;
+    }
+
+    let parent = selectedElement.parent;
+    while (parent) {
+      if (parent.type === TknElementType.PIPELINE_TASK) {
+        return parent as PipelineTask;
+      }
+
+      parent = parent.parent;
+    }
+
+    return undefined;
+
+  }
+
+  private extractTaskDef(taskDoc: VirtualDocument, startPos: number, endPos): string {
+    const task: Task = jsYaml.safeLoad(taskDoc.getText()) as Task;
+    if (!task){
+      throw new Error('Task is empty!');
+    }
+
+    if (task.metadata){
+      if (task.metadata.managedFields){
+        delete task.metadata.managedFields;
+      }
+      if (task.metadata.namespace){
+        delete task.metadata.namespace;
+      }
+      if (task.metadata.resourceVersion){
+        delete task.metadata.resourceVersion;
+      }
+      if (task.metadata.selfLink){
+        delete task.metadata.selfLink;
+      }
+      if (task.metadata.uid){
+        delete task.metadata.uid;
+      }
+      if (task.metadata.generation){
+        delete task.metadata.generation;
+      }
+      if (task.metadata.creationTimestamp){
+        delete task.metadata.creationTimestamp;
+      }
+      if (task.metadata.name){
+        delete task.metadata.name;
+      }
+      if (task.metadata.ownerReferences){
+        delete task.metadata.ownerReferences;
+      }
+      if (task.metadata.annotations['kubectl.kubernetes.io/last-applied-configuration']){
```

Comment:
now getting `Cannot get Tekton Task definition: TypeError: Cannot read property 'kubectl.kubernetes.io/last-applied-configuration' of undefined` error.